packages/shopify-api/docs/guides/graphql-types.md

```diff
@@ -0,0 +1,70 @@
+# Typing GraphQL operations
+
+Since v9, `@shopify/shopify-api` is using `@shopify/admin-api-client` and `@shopify/storefront-api-client` to interact with those APIs.
+
+The main advantage those clients bring are the ability to use [Codegen](https://the-guild.dev/graphql/codegen) to automatically parse and create types for your queries and mutations.
+
+Here is an example of the experience you can get after going through this process:
+
+[typing-graphql-operations.webm](https://github.com/Shopify/shopify-api-js/assets/64600052/a0fed145-b9e1-40b6-9fe0-9d19d9d4c2db)
+
+In this page we'll go over the process of integrating `@shopify/api-codegen-preset` into your app to get all its benefits.
+
+## Set up
+
+The first step is to install the packages:
+
+```bash
+yarn add --dev @shopify/api-codegen-preset
+yarn add @shopify/admin-api-client @shopify/storefront-api-client
+```
+
+> [!NOTE]
+> The app will need to add the clients as direct dependencies so Codegen can overload the types with the ones parsed from your code.
+
+The codegen package includes all the dependencies you'll need to run [`@graphql-codegen/cli`](https://www.npmjs.com/package/@graphql-codegen/cli), including the `graphql-codegen` script.
+
+Then, you can add a `.graphqlrc.ts` file in your app's root.
+See the [package documentation](../../../api-codegen-preset/README.md#configuration) for more details on setting up the appropriate configuration.
```

Comment:
Can we provide a sample one here? Might not be practical since this package is kinda of generic?